devops/scripts/benchmarks/benches/gromacs.py

```diff
@@ -0,0 +1,344 @@
+
+        # Build GROMACS
+        self.build_dir.mkdir(parents=True, exist_ok=True)
+        print(f"Building GROMACS in {self.build_dir}...")
+        subprocess.run(
+            [
+                "cmake",
+                "../",
+                f"-DCMAKE_BUILD_TYPE=Release",
+                f"-DCMAKE_CXX_COMPILER={options.sycl}/bin/clang++",
+                f"-DCMAKE_C_COMPILER={options.sycl}/bin/clang",
+                f"-DGMX_GPU=SYCL",
+                f"-DGMX_SYCL_ENABLE_GRAPHS=ON",
+                f"-DGMX_FFT_LIBRARY=MKL",
+                f"-DGMX_BUILD_OWN_FFTW=ON",
+                f"-DGMX_GPU_FFT_LIBRARY=MKL",
+                f"-DGMX_GPU_NB_CLUSTER_SIZE=8",
+                f"-DGMX_OPENMP=OFF",
+            ],
+            check=True,
+            cwd=self.build_dir,  # Ensure the command runs in the build directory
+        )
+        subprocess.run(["make", "-j"], check=True, cwd=self.build_dir)
+
+        if not self.grappa_dir.exists():
+            self.download_and_extract_grappa()
+        else:
+            print(f"GRAPPA benchmarks already exist at {self.grappa_dir}")
+
+    def download_and_extract_grappa(self):
+        """Download and extract the GRAPPA benchmarks."""
+        grappa_tar_path = self.directory / os.path.basename(self.GRAPPA_BENCHMARKS_URL)
+
+        # Download the GRAPPA tar.gz file
+        if not grappa_tar_path.exists():
+            print(f"Downloading GRAPPA benchmarks from {self.GRAPPA_BENCHMARKS_URL}...")
+            urllib.request.urlretrieve(self.GRAPPA_BENCHMARKS_URL, grappa_tar_path)
+
+        # Extract the GRAPPA tar.gz file
+        print(f"Extracting GRAPPA benchmarks to {self.directory}...")
+        with tarfile.open(grappa_tar_path, "r:gz") as tar:
+            tar.extractall(path=self.directory)
+
+    def teardown(self):
+        print(f"Tearing down GROMACS suite in {self.directory}...")
+        pass
+
+
+class GromacsSystemBenchmark(Benchmark):
+    def __init__(self, suite, system, gromacs_dir, grappa_dir):
+        self.suite = suite
+        self.system = system  # The system name (e.g., "0001.5")
+        self.gromacs_dir = gromacs_dir
+        self.grappa_dir = grappa_dir
+        self.gmx_path = gromacs_dir / "build" / "bin" / "gmx"
+
+    def name(self):
+        return f"gromacs-{self.system}"
+
+    def setup(self):
+        system_dir = self.grappa_dir / self.system
+        if not system_dir.exists():
+            raise FileNotFoundError(f"System directory not found: {system_dir}")
+        print(f"Setting up benchmark for system: {self.system}")
+
+    def run(self, env_vars):
+        if not self.gmx_path.exists():
+            raise FileNotFoundError(f"gmx executable not found at {self.gmx_path}")
+
+        env_vars.update(
+            {
+                "LD_LIBRARY_PATH": f"{options.sycl}/lib"
+                + os.pathsep
+                + os.environ.get("LD_LIBRARY_PATH", ""),
+                "ONEAPI_DEVICE_SELECTOR": "level_zero:gpu",
```

Comment:
`ONEAPI_DEVICE_SELECTOR`  and `SYCL_UR_USE_LEVEL_ZERO_V2` are set from benchmark cli.